tests/func/test_init.py

```diff
@@ -111,3 +111,17 @@ def test_gen_dvcignore(tmp_dir):
         "# https://dvc.org/doc/user-guide/dvcignore\n"
     )
     assert text == (tmp_dir / ".dvcignore").read_text()
+
+
+def test_init_when_ignored_by_git(tmp_dir, scm, monkeypatch, caplog):
+    # https://github.com/iterative/dvc/issues/3738
+    tmp_dir.gen({".gitignore": ".*"})
+    with caplog.at_level(logging.ERROR, logger="dvc"):
+        assert main(["init"]) == 1
```

Comment:
It may be better to test `Repo.init()` here rather than using `main` (since this is an issue with the `init` internals, not necessarily with only the command line `dvc init`).

So something like
```python
    with pytest.raises(InitError):
        # test using DvcRepo.init() instead of main()
        ...
```

Reply:
It may be better to leave testing with `main` and checking `caplog` in this functional test. If only exception is checked when `DvcRepo.init()` is called it may not test the full scenario that happened to users.

In issues

https://github.com/iterative/dvc/issues/3738
https://github.com/iterative/dvc/issues/3803

the users did not see clear message about the issue. Even though the exception when calling the `init` method has that message it could happen in the future that it gets suppressed somehow and is not displayed to users. Testing it with `main` would make it more robust.